### ISSUE TITLE
Localize into more languages

### ISSUE THREAD

**Issue by sr258** [Lumieducation/Lumi#1418]
We should add more languages to Lumi. Here's a list of language codes and language names, roughly sorted by speakers: http://cstrobbe.github.io/languagelearning/misc/languagetags.json It should be easy to add maybe the top 100 of the list (Lumi and h5p-nodejs-library).

We should also think of a way to change the locale of Lumi even if it hasn't been translated into a desired language. This might be useful, because H5P libraries still might have translations, even if Lumi doesn't have those.

**Comment by JPSchellenberg** [comment]
I would suggest to translate into all languages supported by google translate: https://cloud.google.com/translate/docs/languages
Since we are using google translate to translate the locales. It should be no problem run the script for the remaining languages.

**Comment by sr258** [comment]
https://github.com/leolabs/json-autotranslate was released in 1.8.0 which now includes my patches that are required to get it working for our projects. We can now set up a permanent NPM script that translates everything in one go. json-autotranslate also detect new strings and deletes old ones. We should be able to simply run it after having changed a `en.json` file and it takes care of the rest.

**Comment by sr258** [comment]
While we're at it, we should also enable region specific locales. There's an extensive Mexican Spanish localization, and I'm not sure it loads properly. We need to check.

**Comment by DegrangeM** [comment]
The automatic french translation is horrible (especially for words in menu). I wanted to translate it but luckily someone already fixed translations. Thanks pedago44 !

I checked his suggestions and they are all good, you can accept changes. https://translate-h5p.tk/weblate/translate/lumi-education/lumi-app/fr/?q=has:suggestion#suggestions

**Comment by sr258** [comment]
@DegrangeM Thanks for pointing this out. I've missed those suggestions. They're now accepted and already merged into master. The next release will contain them.

**Comment by DegrangeM** [comment]
Is there a way to see new string added by last version to only check new string translation on weblate ?

(also there should be a link to weblate on the contributing part of the readme file)

**Comment by sr258** [comment]
Thanks for pointing this out!

I've added a addon to Weblate that marks new strings a "needs editing". The other H5P components already have this addon, I simply forgot to turn it on for Lumi. From now on, all new strings should become red.

I'll add a link to Weblate!

**Comment by DegrangeM** [comment]
Thanks. Little question about weblate, is there a way to display string retourned by a search in the form of a table ?

When I do a search, result are only displayed one by one and I have to use button to change displayed string:

![image](https://user-images.githubusercontent.com/53106394/115150207-b72a2400-a067-11eb-810f-ac26dd788fcf.png)

Do you know if there is a way to display all results as a table, something like that: 
![image](https://user-images.githubusercontent.com/53106394/115150231-cf01a800-a067-11eb-8662-393c2750d8bf.png)

I searched everywhere but couldn't find but I am surprised weblate doesn't offer this so maybe I missed something. Thanks !

**Comment by sr258** [comment]
Yes, it's called "Zen" mode on the top right. (there's a flash icon next to it)

**Comment by DegrangeM** [comment]
> Yes, it's called "Zen" mode on the top right. (there's a flash icon next to it)

I don't see any flash icon.

![image](https://user-images.githubusercontent.com/53106394/115151905-40912480-a06f-11eb-80bd-fe85485dc254.png)

Maybe it's because I can only add suggestion (but that would be stupid from weblate) ?

**Comment by sr258** [comment]
Yes, you're right, it's probably because it's not possible to add suggestions in Zen mode. I've given you editor rights, so you can work on French to your liking. Problem solved! ;)

Please also push changes under "Manage->Repository maintenance" so that we get a PR on GitHub. That way changes find their way into production quickly.

**Comment by DegrangeM** [comment]
Thanks.

Can you give me more information on this string : https://translate-h5p.tk/weblate/translate/lumi-education/lumi-app/en/?checksum=657d4dbdfcc349bd

I know where it is displayed but don't know what it the meaning of the addon flag of a library.

**Comment by sr258** [comment]
Addons are libraries like H5P.MathDisplay that attach themselves to other libraries and can't be added from the editor by authors.

**Comment by sr258** [comment]
@DegrangeM If you're interested, you can also translate our homepage (https://lumi.education) on Weblate: https://translate-h5p.tk/weblate/projects/lumi-homepage/ Currently it's only auto-translated into French, so a manual go might improve things.

**Comment by DegrangeM** [comment]
The website is actually fine. There is just the Newsletters term that was translated to something nobody use nowdays. I keept the english term as it is often used and understand. I can use a french term if you prefer but this will feel strange I think.

There is an issue in an english string :
> Lumi runs as a desktop program on your computer and it can export HTML files that work on nearly all devices. No need for a **Leraning** Management System like Moodle or a Content Management System like WordPress.

**Comment by sr258** [comment]
Great, thanks. I've corrected mistake in the English source.
